tests/provision/provisioning_rules.go

```diff
@@ -0,0 +1,1334 @@
+package tests
+
+import (
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	mediaSsd       = "SSD"
+	mediaSata      = "SATA"
+	mediaType      = "media_type"
+	liops          = "iops"
+	llatency       = "latency"
+	pxZone         = "failure-domain.beta.kubernetes.io/px_zone"
+	pxRegion       = "failure-domain.beta.kubernetes.io/px_region"
+	kubeZone       = "failure-domain.beta.kubernetes.io/zone"
+	ruleCategories = 6
+)
+
+type labelDict map[string]interface{}
+
+//VpsTemplate provides interface for running individual testcases
+type VpsTemplate interface {
+	// Node label and whether it needs to be set on node remove
+	GetLabels() ([]labelDict, int)
+	// Vps Spec
+	GetSpec() string
+	// Clean up
+	CleanVps()
+}
+
+//VpsRules contains list of all rules for VolumePlacementStrategy
+var VpsRules = []map[string]VpsTemplate{}
+
+// Register registers the given vps rule
+func Register(name string, d VpsTemplate, cat int) error {
+
+	if _, ok := VpsRules[cat][name]; !ok {
+		VpsRules[cat][name] = d
+	} else {
+		return fmt.Errorf("vps rule: %s is already registered", name)
+	}
+	return nil
+}
+
+// GetVpsRules return the list of vps rules
+func GetVpsRules(cat int) map[string]VpsTemplate {
+	if cat >= 0 && cat <= ruleCategories {
+		return VpsRules[cat]
+	}
+	return nil
+
+}
+
+/*
+ *
+ *     Replica  Affinity and Anti-Affinity related test cases
+ *
+ */
+
+type vpscase1 struct {
+	//Case description
+	name string
+	// Enabled
+	enabled bool
+}
+
+//# Case-1--enforcemnt: Required
+func (v *vpscase1) GetLabels() ([]labelDict, int) {
+
+	lbldata := []labelDict{}
+	node1lbl := labelDict{mediaType: mediaSsd}
+	node2lbl := labelDict{mediaType: mediaSata}
+	node3lbl := labelDict{mediaType: mediaSsd}
+	node4lbl := labelDict{mediaType: mediaSsd}
```

Comment:
instead of append, just initial the entire data structure. Makes it easier to read. Same for all places.